### PR TITLE
Toggles password visibility on login form

### PR DESCRIPTION
**Pull request by Skrypt**
orchard

### CODE REVIEW

File: src/OrchardCore.Modules/OrchardCore.Users/Views/Account/Login.cshtml
Comment (agriffard) [inline src/OrchardCore.Modules/OrchardCore.Users/Views/Account/Login.cshtml:43]: Requires font-awesome.

Comment (Skrypt) [reply src/OrchardCore.Modules/OrchardCore.Users/Views/Account/Login.cshtml:43]: Yes

**Comment by Skrypt** [comment]
Don't merge yet I found an issue. For some reason it doesn't switch the icon all the time.

File: src/OrchardCore.Modules/OrchardCore.Users/Views/Account/Login.cshtml
Comment (agriffard) [reply src/OrchardCore.Modules/OrchardCore.Users/Views/Account/Login.cshtml:43]: Then require the Resource or else if the theme doesn't have it, it won't work.

Comment (Skrypt) [reply src/OrchardCore.Modules/OrchardCore.Users/Views/Account/Login.cshtml:44]: done

**Comment by Skrypt** [comment]
Ok so I wanted to use the less dependencies possible on this by using javascript directly.
Did not want either to mess around with adding a function to wait to document to be ready in plain javascript as it can vary between browser versions. I'm handling both cases simply.

File: src/OrchardCore.Modules/OrchardCore.Users/Views/Account/Login.cshtml
Comment (hishamco) [inline src/OrchardCore.Modules/OrchardCore.Users/Views/Account/Login.cshtml:89]: While we have a dependency on jQuery, it will be much simpler if we use it

File: src/OrchardCore.Modules/OrchardCore.Users/Views/Account/Login.cshtml
Comment (hishamco) [inline src/OrchardCore.Modules/OrchardCore.Users/Views/Account/Login.cshtml:104]: It could be done with single condition instead of two

Comment (Skrypt) [reply src/OrchardCore.Modules/OrchardCore.Users/Views/Account/Login.cshtml:104]: I could have chained the conditional operator but I don't do that normally for the sake of simpler readability mainly... not an issue.

File: src/OrchardCore.Modules/OrchardCore.Users/Views/Account/Login.cshtml
Comment (Skrypt) [reply src/OrchardCore.Modules/OrchardCore.Users/Views/Account/Login.cshtml:89]: Well I explained why I did not use it. Now this is just a question of taste at this point. I generally prefer not having to rely on JQuery for everything. If we do this then we should also move this javascript code in the OrchardCore.Users module as a script resource for future extensibility purposes. Though, I'm not sure this is needed unless someone convince me of the opposite. It's just a visual component ...

Comment (hishamco) [reply src/OrchardCore.Modules/OrchardCore.Users/Views/Account/Login.cshtml:89]: No problem, but while there are some UI components are using jQuery heavily seems it needs difficult to rid off the dependency or we may need a time

File: src/OrchardCore.Modules/OrchardCore.Users/Views/Account/Login.cshtml
Comment (deanmarcussen) [inline src/OrchardCore.Modules/OrchardCore.Users/Views/Account/Login.cshtml:94]: cool, lgtm to me.

Only comment is when inlining javascript comments like this they render in the page.

I changed views recently to be razor comments instead, because I felt like we shouldn't be showing our comments to the world, i.e. in the output html.

You might agree / disagree?


Makes sense to me to do this with javascript rather than jquery as this is a view that is often themed, and shouldn't require the theme to use jquery.

**Review by hishamco — APPROVED** [review]
LGTM

File: src/OrchardCore.Modules/OrchardCore.Users/Views/Account/Login.cshtml
Comment (Skrypt) [reply src/OrchardCore.Modules/OrchardCore.Users/Views/Account/Login.cshtml]: Makes sense I changed it to Razor comments

**Comment by deanmarcussen** [comment]
LGTM if everyone is happy with the UI change

**Comment by ns8482e** [comment]
Nice! Can we apply same in Tenant Setup screen?

**Comment by hishamco** [comment]
@ns8482e the icon is fine, the eye should show up with type="password" and eye slash with type="text"

**Comment by ns8482e** [comment]
> @ns8482e the icon is fine, the eye should show up with type="password" and eye slash with type="text"

OK

**Comment by hishamco** [comment]
Perhaps the GIF is confusing little bit but it's right ;)

**Comment by Skrypt** [comment]
> Nice! Can we apply same in Tenant Setup screen?

Done

**Pull request merged**
